send_webhook_payload.py

```diff
@@ -0,0 +1,68 @@
+# send_webhook_payload.py
+# This action extracts metadata from a GitHub Actions event and sends it via HTTP to a database.
+# Note: For now, this action only supports pull request events.
+# Inputs:
+# os.environ['INPUT_COLLECTOR_ENDPOINT']
+# os.environ['INPUT_COLLECTOR_TOKEN']
+# os.environ['INPUT_JIRA_KEYS']
+# os.environ['INPUT_EVENT']
+
+# imports
+import json
+import os
+import re
+import requests
+
+def get_pull_request_info(event_json):
+    pull_request_info = {}
+    pull_request_info['number'] = event_json["pull_request"]["number"]
+    pull_request_info['url'] = event_json["pull_request"]["url"] 
+    pull_request_info['title'] = event_json["pull_request"]["title"]
+    pull_request_info['created_at'] = event_json["pull_request"]["created_at"]
+    pull_request_info['description'] = event_json["pull_request"]["body"]
+    pull_request_info['author'] = event_json["pull_request"]["user"]["login"]
+    return pull_request_info
+
+if __name__=='__main__':
+    # read in env vars
+    collector_endpoint = os.environ['INPUT_COLLECTOR_ENDPOINT']
+    collector_token = os.environ['INPUT_COLLECTOR_TOKEN']
+    jira_keys = os.environ['INPUT_JIRA_KEYS']
+    event = os.environ['INPUT_EVENT']
+
+    # extract all JIRA tickets
+    jira_tickets = []
+    for key in jira_keys.split(","):
+        regex_string = "\b" + key + "-\d+"
+        matches = re.findall(regex_string, event)
+        if matches:
+            jira_tickets.append(matches)
+    # flatten list of lists, if needed
+    all_jira_tickets = [item for sublist in jira_tickets for item in sublist]
+
+    # read in event data into json for remaining queries
+    event_json = json.loads(event)
+
+    # construct payload
+    ## extract general metadata
+    payload = {}
+    payload['repo_org'] = event_json["repository"]["full_name"].split("/")[0]
+    payload['repo_name'] = event_json["repository"]["name"]
+    payload['repo_url'] = event_json["repository"]["html_url"]
+    payload['jira_issues'] = all_jira_tickets
+    ## extract event type-specific information
+    if "pull_request" in event_json.keys():
+        payload['pull_request'] = get_pull_request_info(event_json)
+        payload['event_type'] = "pull_request"
+    else:
+        type_specific_info = {}
+        payload['event_type'] = "unknown"
+
+    # send payload
+    if collector_token:
```

Comment:
Let's add user agent header to all posts

Reply:
It's already included in the default headers: https://github.com/psf/requests/blob/c25af48314c7832c6efb6e5c424253502d779b29/requests/utils.py#L842

Reply:
Ok, what I really mean is I want us to set a custom user agent named after the github action for tracing purposes.